lib/SILOptimizer/Mandatory/DefiniteInitialization.cpp

```diff
@@ -783,6 +786,155 @@ void LifetimeChecker::diagnoseInitError(const DIMemoryUse &Use,
     diagnose(Module, TheMemory.getLoc(), diag::variable_defined_here, isLet);
 }
 
+static void injectHopToExecutorBefore(SILLocation loc,
+                                      SILBasicBlock::iterator &insertPt,
+                                      SILValue actor, bool needsBorrow) {
+  SILBuilder B(insertPt);
+
+  LLVM_DEBUG(llvm::dbgs() << "hop-injector: inserting hop_to_executor before ");
+  LLVM_DEBUG(insertPt->dump());
+
+  if (needsBorrow) {
+    auto *borrow = B.createBeginBorrow(loc, actor);
+    assert(borrow->getNumResults() == 1);
+    actor = borrow->getResult(0);
+  }
+
+  B.createHopToExecutor(loc, actor, /*mandatory=*/false);
+
+  if (needsBorrow)
+    B.createEndBorrow(loc, actor);
+}
+
+void LifetimeChecker::injectActorHops() {
+  auto ctor = TheMemory.isActorInitSelf();
```

Comment:
Please do not return a pointer from an is method. It is misleading since is methods are assumed to return something bool by convention. Keeping in mind that I do not think that you actually need to use this here, for future methods please use a get method, like TheMemory.getActorInitSelf(). If this an is that you really want to use, please return a bool. The reason why I don't think you need the pointer here is the correct thing to use in the inject method (where you are passing this) is just SILBuilderWithScope at the insert pt.

Reply:
That method is used in other places where the pointer is needed. I'll just rename it to use `get` in front instead.